begrun.c

```diff
@@ -51,13 +51,16 @@ void begrun(int RestartSnapNum)
 {
 
     hci_init(HCI_DEFAULT_MANAGER, All.OutputDir, All.TimeLimitCPU, All.AutoSnapshotTime);
-    slots_init();
 
     petaio_init();
     walltime_init(&All.CT);
 
     petaio_read_header(RestartSnapNum);
 
+    int enabled[6] = {All.NTotalInit[0], 0, 0, 0, All.StarformationOn || All.NTotalInit[4], All.BlackHoleOn || All.NTotalInit[5]};
```

Comment:
It might be cleaner to use
```
slots_set_enabled(ptype, elsize)
```
after init.

Reply:
But I would want enabled to be const-after-init if this were C++ :)

Reply:
We probably end up with the convention that struct members are read-only to the outside anyways. So it doesn't really matter. I am in general getting uncomfortable about having an array as an input of a function where the function is only doing a loop over the array (with no cross talk between iterations) -- the interface offers less flexibility than direct accepting the index and defering the loop to the caller.

Reply:
I was tempted to read All.StarFormationOn, etc, inside of slots_init, but that makes the tests more messy. slots_set_enabled is a few more lines of code, but I can use it if you like.

Reply:
I'd say let's go with `slots_set_enabled` or s`lots_enable_type`, and be explicit about the `if-conditions`. It's verbose but cleaner.

And yes we shall try avoid using All.* variables in new code.